src/lib/components/ui/Button/Button.jsx

```diff
@@ -85,37 +79,56 @@ const Button = (props) => {
       >
         {props.label}
       </span>
+      {props.afterLabel && (
+        <span
+          className={`
+            ${styles.afterLabel}
+            ${props.loading ? styles.loadingIcon : ''}
+          `.trim()}
+        >
+          {props.afterLabel}
+        </span>
+      )}
+      {props.endCorner && (
+        <span className={styles.endCorner}>
+          {props.endCorner}
+        </span>
+      )}
     </button>
   );
 };
 
 Button.defaultProps = {
+  afterLabel: null,
+  beforeLabel: null,
   block: false,
   clickHandler: null,
   disabled: false,
-  icon: null,
-  iconPosition: 'before',
+  endCorner: null,
   id: undefined,
   labelVisibility: 'all',
   loading: false,
   priority: 'default',
   size: 'medium',
+  startCorner: null,
   type: 'button',
   variant: 'primary',
 };
 
 Button.propTypes = {
+  afterLabel: PropTypes.element,
+  beforeLabel: PropTypes.element,
   block: PropTypes.bool,
   clickHandler: PropTypes.func,
   disabled: PropTypes.bool,
-  icon: PropTypes.string,
-  iconPosition: PropTypes.oneOf(['before', 'after']),
+  endCorner: PropTypes.element,
   id: PropTypes.string,
   label: PropTypes.string.isRequired,
   labelVisibility: PropTypes.oneOf(['all', 'desktop', 'none']),
   loading: PropTypes.bool,
```

Comment:
I think we should indicate loading status by passing in `loadingIcon` element. This way we would be able to completely separate indication of loading from the other icons. I would prefer to avoid the situation where I have to determine what `afterLabel`/`beforeLabel` icon I have to pass in depending on the loading status.

So I think that `loading: PropTypes.bool,` shoul change to `loadingIcon: PropTypes.element,`.

This way we loose the ability to indicate loading status without displaying icon, but I think that is OK.

Reply:
Makes sense, I second your suggestion.

BTW, I'd like to change to loading status a bit: now the loading icon appears when a button is in loading state which makes the button suddenly grow. This is not a smooth behaviour so I'd like the buttons to retain their original width and reveal the (passed in) loading icon _in place_ of the original label. But I haven't found any time for this and I wanted to finally finish this already lengthy task…